### ISSUE TITLE
Temporarily changing versions for a shell session

### ISSUE THREAD

**Issue by Sevensidedmarble** [asdf-vm/asdf#458]
One thing I really like about pipenv is that it has a command called `pipenv shell` that will open a new shell instance with a given python version set, that's a temporary session.

I would love to have something like this with asdf. Either:
```
asdf shell python 2.7.1
```
That would act like I described above, or something that would not create a new shell process, but would make a change to the shims that would only last for the given shell session (don't know how hard this would be) like:
```
asdf set temp python 2.7.1
```

My reason for wanting this is managing python versioning hell with Neovim. I want to be able to use Python 3.6 for neovim, but 3.5 for everything else by default.

**Comment by Stratus3D** [comment]
This is already supported with environment variables. See https://asdf-vm.github.io/asdf/#/core-manage-versions?id=set-current-version. You'll just need to set `ASDF_${TOOL}_VERSION` before running Vim.

**Comment by avbanks** [comment]
@Stratus3D Can you or someone please provide an example on how to do this with Vim and Python?  I'm struggling to get this to work properly.

**Comment by Stratus3D** [comment]
@avbanks if you have the latest version of asdf installed properly it should be as simple as:

```
$ asdf plugin install python
$ asdf install python 2.7.1
$ asdf shell python 2.7.1
$ echo "$ASDF_PYTHON_VERSION"
2.7.1
$ python # vim, etc... `python` will be 2.7.1
```

**Comment by Startouf** [comment]
asdf shell is no longer available on latest at the current time 0.20.0
